ICSharpCode.ILSpyX/MermaidDiagrammer/ReadMe.md

```diff
@@ -0,0 +1,197 @@
+
+If [**XML documentation comments** are available for the source assembly](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/#create-xml-documentation-output), they're **used to annotate types and members on the generated diagrams**. Commented symbols show up highlighted, making the documentation accessible on hover.
+
+Generated diagrammers offer an **overview** over types, their members and **relations** and the ability to **unfold the domain** along them until you have enough **context** to make an informed decision. Use them as
+- a **mental mapping** tool to get your bearings in an **unknown domain**.
+- a **communication** tool for **your own domain** - when talking about the bigger picture with your team mates or even non-technical shareholders like product owners and users.
+
+# How does it work?
+
+To **extract the type info from the source assembly**, the ilspycmd CLI side-loads it including all its dependencies.
+The current implementation actually uses ILSpy under the hood for that because it's really good at figuring out [which runtime, GAC or private bin path to load referenced assemblies from](https://github.com/icsharpcode/ILSpy/blob/master/ICSharpCode.Decompiler/Metadata/UniversalAssemblyResolver.cs).
+
+The extracted type info is **structured into a model optimized for the HTML diagrammer** and serialized to JSON. The model is a mix between drop-in type definitions in mermaid class diagram syntax and destructured metadata about relations, inheritance and documentation comments.
+
+> The JSON type info is injected into the `template.html` alongside other resources like the `script.js` at corresponding `{{placeholders}}`. It comes baked into the HTML diagrammer to enable
+> - accessing the data and
+> - importing the mermaid module from a CDN
+>
+> locally without running a web server [while also avoiding CORS restrictions.](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy#file_origins)
+>
+> Note that you can
+> - **customize** the `template.html`, `script.js` and `styles.css` used in this process, e.g. for rewording or branding the UI.
+> - **build a stand-alone offline diagrammer** with *mermaid* baked in instead of loading it from a CDN. All you have to do is replace the `import` statement for `/dist/mermaid.esm.min.mjs` at the top of `script.js` with the *contents* of [`/dist/mermaid.min.js`](https://unpkg.com/mermaid@latest/dist/mermaid.min.js) from the *mermaid* package of your choice.
+
+In the final step, the **HTML diagrammer app re-assembles the type info** based on the in-app type selection and rendering options **to generate [mermaid class diagrams](https://mermaid.js.org/syntax/classDiagram.html)** with the types, their relations and as much inheritance detail as you need.
+
+# Check out the demo
```

Comment:
Can we use any of the icsharpcode repos as the demo going forward? Note that we https://github.com/icsharpcode/ILSpy/tree/gh-pages hosted as ilspy.net (only / is redirecting to the repo, other pages do work) which might make hosting the demo(s) nicer and easier.

Reply:
Sure you can. I've had a quick look at `gh-pages` and think I get the idea. Do I create a separate PR for that then or how does the update process work?

Reply:
E.g. I've pushed a demo diagrammer to https://github.com/h0lg/ILSpy/tree/gh-pages-update

Reply:
Yes, that is a separate branch that will need a separate PR

Reply:
I've replace the demo links.